libflux/core/src/semantic/convert.rs

```diff
@@ -242,42 +242,311 @@ fn convert_monotype(
 ) -> Result<types::PolyType> {
     let mut tvars = HashMap::<String, types::Tvar>::new();
-    let expr = convert_monotype(type_expression.monotype, &mut tvars, f)?;
+    let mut tvars_mapped = HashMap::<types::Tvar, types::Tvar>::new();
+    let mut expr = convert_monotype(type_expression.monotype, &mut tvars, f)?;
     let mut vars = Vec::<types::Tvar>::new();
-    let mut cons = SemanticMap::<types::Tvar, Vec<types::Kind>>::new();
-
-    for (name, tvar) in tvars {
-        vars.push(tvar);
-        let mut kinds = Vec::<types::Kind>::new();
+    let mut recs = HashMap::<types::Tvar, MonoType>::new();
+    
+    let row_tvar = f.fresh();
+    vars.push(row_tvar);
+    for (name, tvar) in &tvars {
+        let mut rec = MonoType::Var(row_tvar);
+        let new_tvar = f.fresh();
+        tvars_mapped.insert(tvar.clone(), new_tvar);
+        vars.push(new_tvar);
         for con in &type_expression.constraints {
-            if con.tvar.name == name {
+            if con.tvar.name == *name {
                 for k in &con.kinds {
                     match k.name.as_str() {
-                        "Addable" => kinds.push(types::Kind::Addable),
-                        "Subtractable" => kinds.push(types::Kind::Subtractable),
-                        "Divisible" => kinds.push(types::Kind::Divisible),
-                        "Numeric" => kinds.push(types::Kind::Numeric),
-                        "Comparable" => kinds.push(types::Kind::Comparable),
-                        "Equatable" => kinds.push(types::Kind::Equatable),
-                        "Nullable" => kinds.push(types::Kind::Nullable),
-                        "Negatable" => kinds.push(types::Kind::Negatable),
-                        "Timeable" => kinds.push(types::Kind::Timeable),
-                        "Record" => kinds.push(types::Kind::Record),
+                        "Addable" => {
+                            rec = MonoType::Record(Box::new(types::Record::Extension {
+                                head: types::Property {
+                                    k: "+".to_string(),
+                                    v: MonoType::Var(new_tvar),
+                                },
+                                tail: rec
+                            }))
+                        }
+                        "Subtractable" => {
+                            rec = MonoType::Record(Box::new(types::Record::Extension {
+                                head: types::Property {
+                                    k: "-".to_string(),
+                                    v: MonoType::Var(new_tvar),
+                                },
+                                tail: rec
+                            }))
+                        }
+                        "Divisible" => {
+                            rec = MonoType::Record(Box::new(types::Record::Extension {
+                                head: types::Property {
+                                    k: "*".to_string(),
```

Comment:
Do we need to include all these labels for `Divisible`? Will we run into any issues if the only label is `/` similar to the `Addable` constraint?